llvm/lib/CodeGen/TargetLoweringObjectFileImpl.cpp

```diff
@@ -2378,13 +2394,28 @@ MCSection *TargetLoweringObjectFileXCOFF::getExplicitSectionGlobal(
   StringRef SectionName = GO->getSection();
```

Comment:
See `report_fatal_error` above.
I think the new code should be moved up so that `SectionName` is correct by the time we reach the next line.

I also believe that this `pragma` should suppress the application of `-mtocdata`.
For explicit section attributes and `-mtocdata=<var>`, there is a warning implemented:
```
-mtocdata option is ignored for <var> because variable has a section attribute
```

Reply:
The focus was from the beginning on COFF format. We misinterpreted and thought the XCOFF would be the same way. It doesn't look to be so, so we decided to drop the changes for XCOFF from this PR. This also makes this PR more isolated.